src/state/trajectory/logics.ts

```diff
@@ -640,11 +652,30 @@ const cancelConversionLogic = createLogic({
     type: CANCEL_CONVERSION,
 });
 
-const setInitialSelectedUIData = createLogic({
-    process(deps: ReduxLogicDeps, dispatch) {
-        const { action } = deps;
-        const uiData = action.payload;
-        dispatch(setSelectedUIDisplayData(uiData));
+const receiveDefaultUIDataLogic = createLogic({
+    process(deps: ReduxLogicDeps, dispatch, done) {
+        const { getState, action } = deps;
+        const browserStoredUIData = getSelectedUIDisplayData(getState());
+        /**
+         * If the conditions below are true, then valid color settings were
+         * retrieved from browser storage before the default ui data arrived
+         * and those settings should be applied.
+         * If false then userSelected data either doesn't exist, hasn't arrived yet, or
+         * is invalid: set to empty array for now.
+         */
+        if (
+            browserStoredUIData.length > 0 &&
+            compareAgentTrees(browserStoredUIData, action.payload)
```

Comment:
I would rename this to `isSameAgentTree` since you're using it as a bool and the naming is ambiguous